lib/boss_rds.py

```diff
@@ -0,0 +1,213 @@
+        finally:
+            sql.close()
+
+def get_mysql_params(session, domain):
+    """
+    Get MySQL connection info from Vault.
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+
+    Returns:
+        (DbParams): Connection info from Vault.
+    """
+    keypair = aws.keypair_lookup(session)
+    call = ExternalCalls(session, keypair, domain)
+    names = AWSNames(domain)
+    DB_HOST_NAME = names.endpoint_db.split(".")[0]
+    logging.debug("DB Hostname is: {}".format(DB_HOST_NAME))
+
+    logging.info('Getting MySQL parameters from Vault (slow) . . .')
+    with call.vault() as vault:
+        params = vault.read('secret/endpoint/django/db')
+
+    return DbParams('{}.{}'.format(DB_HOST_NAME, domain),params['port'],params['name'], params['user'], params['password'])
+
+def sql_list(session, domain, db_table):
+    """
+    List all the available members of a given sql table.
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+        db_table: Identifies which table members to list.
+
+    Returns:
+        (str): Lookup key.
+    """
+    query = "SELECT * FROM {}".format(db_table)
+
+    with connect_rds(session, domain) as sql:
+        try:
+            cursor = sql.cursor()
+            cursor.execute(query)
+            ans = cursor.fetchall()
+            for i in ans:
+                logging.info(i)
+        finally:
+            cursor.close()
+
+def sql_resource_lookup_key(session, domain, resource_params):
+    """
+    Get the lookup key that identifies the resource from the database.
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+        resource_params (str): Identifies collection, experiment or channel.
+
+    Returns:
+        (str): Lookup key.
+    """
+    collection, experiment, channel = None, None, None
+    resource = resource_params.split("/")
+    
+    if len(resource) == 0:
+        raise Exception("Incorrect number of arguments(Make sure the resource provided has at least a collection to lookup)")
+    else:
+        if len(resource) > 0:
+            collection = resource_params.split("/")[0]
+        if len(resource) > 1:
+            experiment = resource_params.split("/")[1]
+        if len(resource) > 2:
+            channel = resource_params.split("/")[2]
+        elif len(resource) > 3:
+            raise Exception("Only provide /coll/exp/chan")
+
+    coll_query = "SELECT id FROM collection WHERE name = %s"
+    exp_query = "SELECT id FROM experiment WHERE name = %s"
+    chan_query = "SELECT id FROM channel WHERE name = %s"
+
+    with connect_rds(session, domain) as sql:
+        try:
+            cursor = sql.cursor()
+            if collection is not None:
+                cursor.execute(coll_query, (collection,))
+                coll_set = cursor.fetchall()
+                if len(coll_set) != 1:
```

Comment:
Could create a helper method like
```py
def check_result(rst, resource, name):
    if len(rst) != 1:
        raise ResourceNotFoundException("... {}: {}".format(resource, name)
```
which would replace the `if / else` with a single line call, making it a little easier to read the code